app/Http/Requests/Admin/StoreOrganizationType.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace App\Http\Requests\Admin;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class StoreOrganizationType extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        $orgType = $this->route('organization_type');
+        if($orgType){
+            return [
+                'name' => 'required|string|max:255|unique:organization_types,name,'.$orgType->id,
+                'label' => 'required|string|max:255|unique:organization_types,label,'.$orgType->id,
```

Comment:
Do we need $orgType->id for unique?

Reply:
Also I think name and label are not required for unique.

Reply:
We need the id for update scenarios. Example:

I have
{
label: 'Other',
name:'other'
}

And I want to change it to 

{
label: 'Other Orgs',
name:'other'
}

This will fail validation because name is unique and you're sending the same name. Laravel provides this feature to ignore these unique fields only in the row where the ID is the same, as used in the code here.

As for being unique: name is supposed to be the machine name for the org type, to be used in internal queries. This means it has to be unique. Label could conceptually be repeatable but in practice we're storing full text org types in the user table. To maintain compatibility with that, we're sending the label value in the registration form instead of the name, effectively making it the index. So yeah, that means it should be unique too.

Uniqueness is also important conceptually because this is an enumeration type of field to be used in a select box in registration. Having duplicates will defeat its classification purpose.

Reply:
So does it only validate for those same id, name and label pair?

Reply:
No, the other way around: The "unique" validation is ignored for the row with the matching ID. It allows it to be equal to itself but not equal to others. The rest of the validation (req, string, max) happens for all cases.